builtin/providers/ultradns/resource_ultradns_rdpool.go

```diff
@@ -0,0 +1,239 @@
+	_, err = client.RRSets.Create(r.RRSetKey(), r.RRSet())
+	if err != nil {
+		return fmt.Errorf("create failed: %#v -> %v", r, err)
+	}
+
+	d.SetId(r.ID())
+	log.Printf("[INFO] ultradns_rdpool.id: %v", d.Id())
+
+	return resourceUltradnsRdpoolRead(d, meta)
+}
+
+func resourceUltradnsRdpoolRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] ultradns_rdpool read")
+	client := meta.(*udnssdk.Client)
+
+	rr, err := newRRSetResourceFromRdpool(d)
+	if err != nil {
+		return err
+	}
+
+	rrsets, err := client.RRSets.Select(rr.RRSetKey())
+	if err != nil {
+		uderr, ok := err.(*udnssdk.ErrorResponseList)
+		if ok {
+			for _, resps := range uderr.Responses {
+				// 70002 means Records Not Found
+				if resps.ErrorCode == 70002 {
+					d.SetId("")
+					return nil
+				}
+				return fmt.Errorf("resource not found: %v", err)
+			}
+		}
+		return fmt.Errorf("resource not found: %v", err)
+	}
+
+	r := rrsets[0]
+
+	zone := d.Get("zone")
+	// ttl
+	d.Set("ttl", r.TTL)
+	// hostname
+	if r.OwnerName == "" {
+		d.Set("hostname", zone)
+	} else {
+		if strings.HasSuffix(r.OwnerName, ".") {
+			d.Set("hostname", r.OwnerName)
+		} else {
+			d.Set("hostname", fmt.Sprintf("%s.%s", r.OwnerName, zone))
+		}
+	}
+
+	// And now... the Profile!
+	if r.Profile == nil {
+		return fmt.Errorf("RRSet.profile missing: invalid RDPool schema in: %#v", r)
+	}
+	p, err := r.Profile.RDPoolProfile()
+	if err != nil {
+		return fmt.Errorf("RRSet.profile could not be unmarshalled: %v\n", err)
+	}
+
+	// Set simple values
+	d.Set("description", p.Description)
+	d.Set("order", p.Order)
+
+	err = d.Set("rdata", makeSetFromStrings(r.RData))
+	//err = d.Set("rdata", makeSetFromRdataAlone(r.RData))
+	if err != nil {
+		return fmt.Errorf("rdata set failed: %#v", err)
+	}
+	return nil
+}
+
+func resourceUltradnsRdpoolUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] ultradns_rdpool update")
+	client := meta.(*udnssdk.Client)
+
+	r, err := newRRSetResourceFromRdpool(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] ultradns_rdpool update: %+v", r)
+	_, err = client.RRSets.Update(r.RRSetKey(), r.RRSet())
+	if err != nil {
+		return fmt.Errorf("resource update failed: %v", err)
+	}
+
+	return resourceUltradnsRdpoolRead(d, meta)
+}
+
+func resourceUltradnsRdpoolDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] ultradns_rdpool delete")
+	client := meta.(*udnssdk.Client)
+
+	r, err := newRRSetResourceFromRdpool(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[INFO] ultradns_rdpool delete: %+v", r)
+	_, err = client.RRSets.Delete(r.RRSetKey())
+	if err != nil {
+		return fmt.Errorf("resource delete failed: %v", err)
+	}
+
+	return nil
+}
+
+// Resource Helpers
+
+func newRRSetResourceFromRdpool(d *schema.ResourceData) (rRSetResource, error) {
+	//rDataRaw := d.Get("rdata").(*schema.Set).List()
+	r := rRSetResource{
+		// "The only valid rrtype value for SiteBacker or Traffic Controller pools is A"
```

Comment:
I guess it's implied that since this isn't a SiteBacker or Traffic Controller pool, we're inferring the same semantics apply to Resource Distribution pools.